src/NBitcoin/BitcoinSegwitAddress.cs

```diff
@@ -28,16 +35,15 @@ private static string Validate(string bech32, ref Network expectedNetwork)
                 {
                     byte witVersion;
                     var data = encoder.Decode(bech32, out witVersion);
-                    if(data.Length == 20 && witVersion == 0)
+                    if (data.Length == 20 && witVersion == 0)
                     {
-                        expectedNetwork = network;
-                        return bech32;
+                        return true;
                     }
                 }
-                catch(Bech32FormatException) { throw; }
-                catch(FormatException) { continue; }
+                catch (Bech32FormatException) { throw; }
```

Comment:
redundant line ` catch (Bech32FormatException) { throw; }`

Reply:
I assumed you were talking about the one-after-another `FormatException`s, which have been removed.